endorsement/dao/shared_drive.py

```diff
@@ -1,65 +1,99 @@
+from endorsement.dao.gws import is_group_member
+from endorsement.dao.itbill import get_subscription_by_key_remote
+from endorsement.exceptions import ITBillSubscriptionNotFound
+from endorsement.models import (
+    ITBillSubscription,
+)
+
 from endorsement.models.shared_drive import (
-    SharedDriveMember, Member, Role, SharedDrive,
-    SharedDriveQuota, SharedDriveRecord)
+    Member,
+    Role,
+    SharedDrive,
+    SharedDriveMember,
+    SharedDriveQuota,
+    SharedDriveRecord,
+)
 from endorsement.exceptions import SharedDriveNonPrivilegedMember
-import csv
 import re
 import logging
 
+# methods are imported to populate namespace and be used elsewhere
+from uw_msca.shared_drive import (  # noqa: F401
+    get_default_quota,
+    get_google_drive_states,
+    set_drive_quota,
+)
+from uw_msca.models import GoogleDriveState
+
 
 logger = logging.getLogger(__name__)
 netid_regex = re.compile(
-    r'^(?P<netid>[^@]+)@(uw|(u\.)?washington)\.edu$', re.I)
+    r"^(?P<netid>[^@]+)@(uw|(u\.)?washington)\.edu$", re.I
+)
 
 
-def load_shared_drives_from_csv(file_path):
+def load_shared_drives(google_drive_states):
     """
-    populate shared drive models
+    Populate SharedDriveRecord models as reported by uw_mwsa.
+
+    Returns the final SharedDriveRecord created.
     """
-    seen_shared_drives = set()
-    columns = None
-    with open(file_path, 'r') as csvfile:
-        for row in csv.reader(csvfile, delimiter=","):
-            try:
-                if columns:
-                    a = DataFromRow(row=row, columns=columns)
-                    load_shared_drive_record(
-                        a, a.DriveId in seen_shared_drives)
-                    seen_shared_drives.add(a.DriveId)
-                else:
-                    columns = list(filter(None, row))
-            except SharedDriveNonPrivilegedMember as ex:
-                logger.info(f"{ex}")
-            except Exception as ex:
-                logger.error(f"shared drive record: {a}: {ex}")
+    seen_drive_ids = set()
 
+    retval = None
 
-def load_shared_drive_record(a, is_seen):
+    for gds in google_drive_states:
+        try:
+            retval: SharedDriveRecord = load_shared_drive_record(
+                gds,
+                is_seen=gds.drive_id in seen_drive_ids,
+            )
+        except SharedDriveNonPrivilegedMember as e:
+            logger.info(f"{e}")
+        except Exception as e:
+            logger.error(f"shared drive record: {gds}: {e}")
+        else:
+            seen_drive_ids.add(gds.drive_id)
+
+    return retval
+
+
+def load_shared_drive_record(a: GoogleDriveState, is_seen):
     """
     ensure shared drive record is created
     """
     shared_drive = upsert_shared_drive(a, is_seen)
     shared_drive_record, _ = SharedDriveRecord.objects.get_or_create(
-        shared_drive=shared_drive)
+        shared_drive=shared_drive,
+        defaults={
+            "datetime_created": dt.datetime.now(dt.timezone.utc),
```

Comment:
@mikeseibel Please confirm `datetime_created` is meant to refer to the creation of the `SharedDriveRecord` object.